mlflow/spark.py

```diff
@@ -909,6 +988,12 @@ def predict(
         """
         from pyspark.ml import PipelineModel
 
+        if _is_spark_connect_model(self.spark_model):
+            # Spark connect ML model directly append prediction result column to input pandas
+            # dataframe. To make input dataframe intact, make a copy first.
```

Comment:
Does this mean `Spark connect ML model` mutates the passed-in pandas dataframe?

Reply:
Yes.

Reply:
Got it, why did we make this decision?

Reply:
In design stage, we want to avoid data copying, 
For `pyspark.ml` model, it returns new spark dataframe with new column, but spark dataframe is lazy computed,
pandas dataframe is different, when appending new column we have to construct dataframe content.

Reply:
Is there a method that only return prediction values?

Reply:
> Is there a method that only return prediction values?

Currently no. Spark connect model is designed to have similar input / output style with legacy spark model:

For legacy spark model, input is a spark dataframe and output is a spark dataframe containing prediction result columns,

For spark connect model, input / output are either spark dataframe or pandas dataframe, but they have similar columns with legacy spark model input / output

Reply:
Yep! From a user perspective, this implementation mirrors that of current SparkML :)